Posts & Pages: Add status text

kean opened the pull request:
- Display status of the posts
- Fix spacing for excerpts

<img width="320" alt="Screenshot 2023-10-26 at 8 43 31 AM" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/c11519d1-8d30-4ea9-b5fc-841ce63b7e2a">
<img width="320" alt="Screenshot 2023-10-26 at 8 44 59 AM" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/a8cf4f63-d1c9-4096-bcbe-bb866b680666">
<img width="320" alt="Screenshot 2023-10-26 at 8 43 34 AM" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/0d10baec-0569-4cfa-994f-44b1b8fcf6c9">
<img width="320" alt="Screenshot 2023-10-26 at 8 43 36 AM" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/c53db55c-3faf-4b30-b233-47f9a6001ecc">


## Regression Notes
1. Potential unintended areas of impact: Posts & Pages
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

kean committed:
Display post status

kean committed:
Display status with date for pages

kean committed:
Fix spacing for excerpts

kean committed:
Switch to dateModified to trashed

momo-ozawa commented on WordPress/Classes/ViewRelated/Post/AbstractPostHelper.swift:
```diff
@@ -0,0 +1,39 @@
+    static let scheduled = NSLocalizedString("post.scheduledForDate", value: "Scheduled %@", comment: "Post status and date for list cells")
+    static let created = NSLocalizedString("post.createdTimeAgo", value: "Created %@", comment: "Post status and date for list cells")
+    static let edited = NSLocalizedString("post.editedTimeAgo", value: "Edited %@", comment: "Post status and date for list cells")
+    static let trashed = NSLocalizedString("post.trashedTimeAgo", value: "Trashed %@", comment: "Post status and date for list cells")
```
In the comments, let's add "%@ is a placeholder for the date.", to add more context for our translators.

momo-ozawa commented on WordPress/Classes/ViewRelated/Post/PostListHeaderView.swift:
```diff
@@ -60,6 +61,8 @@ final class PostListHeaderView: UIView {
         setupLabelStackView()
         setupEllipsisButton()
 
+        dateLabel.setContentCompressionResistancePriority(.init(rawValue: 800), for: .horizontal)
```
I wonder if it would be better to convert the header labels to a single attributed string, like in the pages cell. In setupLabelStackView(), I'm setting the content hugging priority to high to prevent labels from going (except for the last label), which seems overly complicated in retrospect

momo-ozawa reviewed (commented):
On the trashed tab, the posts are ordered by date created instead of date modified

<img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/27b44dd6-79e9-4146-a7ec-cf054772810d" width=200>

kean committed:
Use attributed string for post badges

kean committed:
Update l10n

kean committed:
Remove isAuthorBadgeHidden

kean commented:
Changes since the last update:

- Switched to NSAttributedString for post badges
- Fixed an issue where the authors label was shown
- Fixed the trashed list by showing `dateModified` instesfd of `dateCreated`. Based on my testing, when you trash a post, the `modified` date gets updates. Presumably, you can't alter the post any further.
